src/coreclr/src/vm/ecalllist.h

```diff
@@ -529,66 +529,87 @@ FCFuncStart(gDelegateFuncs)
     FCFuncElement("DelegateConstruct", COMDelegate::DelegateConstruct)
 FCFuncEnd()
 
+#ifdef _MSC_VER && defined(TARGET_AMD64)
```

Comment:
I do not think you need to do the `#pragma float_control` here. It should not be required to workaround the bug.

Reply:
Sorry I don't understand. What is the workaround if it is not float_control?

Reply:
The problem is that `fma` implementation in Windows ARM64 C runtime library is slow and broken. `fma` is is treated as intrinsic by the C++ compiler by default, so this broken `fma` implementation is used very rarely.

When you wrap the `fma` in a helper method, the intrinsic implementation kicks in and everything works correctly. The broken `fma` implementation does not get used.

`float_control on/off` does not make any difference

Reply:
Thank you for the explanation. I have changed mathfuncs.h as you have suggested below.